build_daemon/lib/src/server.dart

```diff
@@ -90,7 +92,11 @@ class Server {
     });
 
     var server = _server = await HttpMultiServer.loopback(0);
-    serveRequests(server, handler);
+    // Serve requests in an error zone to prevent failures
+    // when running from another error zone.
+    runZonedGuarded(() => serveRequests(server, handler), (e, _) {
+      _logger.warning('Error serving requests: $e');
```

Comment:
doesn't the logger have an exception and stack argument?

Reply:
Yes, I would pass those assuming we properly log them for warnings but I think we should (or we should update the logging code if not).

Reply:
done.